src/engine/interop/specification.md

```diff
@@ -0,0 +1,130 @@
+
+### engine_executePayload
+
+#### Parameters
+1. `Object` - Instance of [`ExecutionPayload`](#ExecutionPayload)
+
+#### Returns
+`Object` - Response object:
+1. `status`: `String` - the result of the payload execution:
+  - `VALID` - given payload is valid
+  - `INVALID` - given payload is invalid
+  - `SYNCING` - sync process is in progress
+
+#### Specification
+
+1. Client software **MUST** validate the payload according to the execution environment rule set defined in the [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and respond with the validation result.
+
+2. Client software **MUST** defer persisting a valid payload until the corresponding [**`engine_consensusValidated`**](#engine_consensusValidated) message deems the payload valid with respect to the proof-of-stake consensus rules.
+
+3. Client software **MUST** discard the payload if it's deemed invalid.
+
+4. The call **MUST** be responded with `SYNCING` status while the sync process is in progress and thus the execution cannot yet be validated.
+
+5. In the case when the parent block is unknown, client sofware **MUST** pull the block from the network and take one of the following actions depending on the parent block properties:
+  - If the parent block is a PoW block as per [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) definition then all missing dependencies of the payload **MUST** be pulled from the network and validated accordingly. The call **MUST** be responded according to the validity of the payload and the chain of its ancestors.
+  - If the parent block is a PoS block as per [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) definition then the call **MUST** be responded with `SYNCING` status and sync process **SHOULD** be initiated as it is specified in the [merge-sync.md](https://github.com/fjl/p2p-drafts/blob/master/merge-sync/merge-sync.md) document.
+
+### engine_consensusValidated
+
+#### Parameters
+1. `Object` - Payload validity status with respect to the consensus rules:
+- `blockHash`: `DATA`, 32 Bytes - block hash value of the payload
+- `status`: `String: VALID|INVALID` - result of the payload validation with respect to the proof-of-stake consensus rules
+
+#### Returns
+none
+
+#### Specification
+
+1. The call of this method with `VALID` status maps on the `POS_CONSENSUS_VALIDATED` event of [EIP-3675](https://eips.ethereum.org/EIPS/eip-3675#specification) and **MUST** be processed according to the specification defined in the EIP.
+
+2. If the status in this call is `INVALID` the payload **MUST** be discarded disregarding its validity with respect to the execution environment rules.
+
+### engine_forkchoiceUpdated
+
+#### Parameters
+1. `Object` - The state of the fork choice:
+- `headBlockHash`: `DATA`, 32 Bytes - block hash of the head of the canonical chain
+- `finalizedBlockHash`: `DATA`, 32 Bytes - block hash of the most recent finalized block
+
+#### Returns
+none
```

Comment:
What if the el engine does not have the `finalizedBlockHash` in db? 
Previously `SetHead` returned `Success:False` for that

Reply:
Good question. I think then the sync process should be initiated (if it hasn't been previously, this could be the first message after a start either with a fresh state or after a long period of being out) and the corresponding `SYNCING` response should be sent back. WDYT?

Reply:
cc @fjl

Reply:
I was thinking about the design of the `forkchoiceUpdated` for a little bit. IMO, it should be designed as a notification hence no response is expected. The same should be applicable to the `consensusValidated`. 

I also think that we better use `executePayload` as the only method related to sync process if we want sync to piggy back on the core methods and responses to them. A slight improvement here could be an introduction of another type of response to the `executePayload` which provides more detail on the sync process.